lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php

```diff
@@ -151,7 +151,11 @@ public function lastInsertId($name = null)
             return false;
         }
 
-        OraclePlatform::assertValidIdentifier($name);
+        $identifiers = explode('.', $name);
+
+        foreach ($identifiers as $identifier) {
```

Comment:
Okay thought let's see what `PDO_OCI` has to tell about this thing:

```
deeky@98N-MUELLER-STEVE:~/dev/doctrine/dbal$ vendor/bin/phpunit -c pdo_oci.phpunit.xml tests/Doctrine/Tests/DBAL/Functional/Driver/PDOOracle/ConnectionTest.php 
PHPUnit 5.4.6 by Sebastian Bergmann and contributors.

E                                                                   1 / 1 (100%)

Time: 214 ms, Memory: 4.00MB

There was 1 error:

1) Doctrine\Tests\DBAL\Functional\Driver\PDOOracle\ConnectionTest::testLastInsertIdAcceptsFqn
Exception: [PDOException] SQLSTATE[IM001]: Driver does not support this function: driver does not support lastInsertId()

With queries:
4. SQL: 'INSERT INTO DBAL2595 (foo) VALUES (1)' Params: 
3. SQL: 'CREATE TABLE DBAL2595(id NUMBER DEFAULT dbal2595_seq.NEXTVAL, foo NUMBER)' Params: 
2. SQL: 'CREATE SEQUENCE dbal2595_seq' Params: 
1. SQL: 'ALTER SESSION SET NLS_TIME_FORMAT = 'HH24:MI:SS' NLS_DATE_FORMAT = 'YYYY-MM-DD HH24:MI:SS' NLS_TIMESTAMP_FORMAT = 'YYYY-MM-DD HH24:MI:SS' NLS_TIMESTAMP_TZ_FORMAT = 'YYYY-MM-DD HH24:MI:SS TZH:TZM' NLS_NUMERIC_CHARACTERS = '.,'' Params: 

Trace:
/home/deeky/dev/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:123
/home/deeky/dev/doctrine/dbal/tests/Doctrine/Tests/DBAL/Functional/Driver/PDOOracle/ConnectionTest.php:41
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:1081
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:932
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/src/Framework/TestResult.php:701
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/src/Framework/TestCase.php:888
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/src/Framework/TestSuite.php:753
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:465
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/src/TextUI/Command.php:162
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/src/TextUI/Command.php:113
/home/deeky/dev/doctrine/dbal/vendor/phpunit/phpunit/phpunit:47


/home/deeky/dev/doctrine/dbal/tests/Doctrine/Tests/DbalFunctionalTestCase.php:81

Caused by
PDOException: SQLSTATE[IM001]: Driver does not support this function: driver does not support lastInsertId()

/home/deeky/dev/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:123
/home/deeky/dev/doctrine/dbal/tests/Doctrine/Tests/DBAL/Functional/Driver/PDOOracle/ConnectionTest.php:41

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```

Nuff said... :D